File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
  h = 0
  if #a ~= #b then
    return -1
  else
    for i = 1, #a do
      if string.sub(a, i, i) ~= string.sub(b, i, i) then
        h = h + 1
      end
    end
  end
  return h
end

return Hamming

File: fortran/hamming/hamming.f90
MODULE hamming
  IMPLICIT NONE
CONTAINS

  FUNCTION compute(strand1, strand2, distance)
    CHARACTER(*) :: strand1, strand2
    INTEGER :: distance, i
    LOGICAL :: compute

    distance = 0
    compute = .FALSE.

    IF ( LEN(strand1) /= LEN(strand2) ) THEN
       RETURN
    END IF
    DO i = 1, LEN(strand1), 1
       IF ( strand1(i:i) /= strand2(i:i) ) THEN
          distance = distance + 1
       END IF
    END DO
    compute = .TRUE.
    RETURN
  END FUNCTION compute

END MODULE hamming
